### PR TITLE
Fix query plan cache initialization

### PR DESCRIPTION
**Pull request by kul**
There is a 'memory leak' introduced due to below commit as the init(..) code path (via sync clients) does not initializes the cache to a size limited lru cache. In case the queries are not parameterized this can lead to the cache growing unbounded over time.

https://github.com/Azure/azure-sdk-for-java/commit/01ad248c46763435bbce35750e23297a46070f2d#diff-dd557c9356c91a57b6bc7ce3cff436e1fde4daaa2841a2de889b3a7da138afb4

### CODE REVIEW

**Comment by kushagraThapar** [comment]
@mbhaskar  - can you please take a look at this.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java [hunk 1]
```diff
@@ -467,7 +467,7 @@ public void init(CosmosClientMetadataCachesSnapshot metadataCachesSnapshot, Func
                 connectionPolicy.getConnectionMode(), globalEndpointManager.getLatestDatabaseAccount().getId(),
                 null, null, this.reactorHttpClient, connectionPolicy.isClientTelemetryEnabled(), this, this.connectionPolicy.getPreferredRegions());
             clientTelemetry.init();
-            this.queryPlanCache = new ConcurrentHashMap<>();
+            this.queryPlanCache = Collections.synchronizedMap(new SizeLimitingLRUCache(Constants.QUERYPLAN_CACHE_SIZE));
```
Comment (mbhaskar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java:470]: I feel we can skip initializing this here as it gets initialized in the client construction before init here: https://github.com/Azure/azure-sdk-for-java/blob/ff077373038ff09cf967ddab3d3cf4f0649ca60a/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java#L230 -> 
https://github.com/Azure/azure-sdk-for-java/blob/ff077373038ff09cf967ddab3d3cf4f0649ca60a/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java#L373

So the fix just would be deleting the line that initializes queryPlanCache here

```suggestion

```

**Review by mbhaskar — CHANGES_REQUESTED** [review]
Thanks @kul for identifying this. Left some review comments

**Review by mbhaskar — APPROVED** [review]
LGTM. Thanks. 
Please resolve conflicts and also make sure tests pass.

**Pull request merged**